refurb/checks/datetime/unreliable_utc_usage.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from typing import Final
+
+from mypy.nodes import CallExpr, MemberExpr, NameExpr
+
+from refurb.error import Error
+from refurb.settings import Settings
+
+
+@dataclass
+class ErrorInfo(Error):
+    """
+    Because naive `datetime` objects are treated by many `datetime` methods
+    as local times, it is preferred to use aware datetimes to represent times
+    in UTC.
+
+    This check affects `datetime.utcnow` and `datetime.utcfromtimestamp`.
+
+    Bad:
+
+    ```
+    from datetime import datetime
+
+    now = datetime.utcnow()
+    past_date = datetime.utcfromtimestamp(some_timestamp)
+    ```
+
+    Good:
+
+    ```
+    from datetime import datetime, timezone
+
+    datetime.now(timezone.utc)
+    datetime.fromtimestamp(some_timestamp, tz=timezone.utc)
+    ```
+    """
+
+    name = "unreliable-utc-usage"
+    code = 176
+    categories = ("datetime",)
+
+
+_replacements: Final = {
+    "utcnow": ("()", "now(tz=timezone.utc)"),
+    "utcfromtimestamp": ("(...)", "fromtimestamp(..., tz=timezone.utc)"),
+}
+
+
+def check(node: CallExpr, errors: list[Error], settings: Settings) -> None:
+    match node:
+        case CallExpr(
+            callee=MemberExpr(
+                expr=NameExpr(fullname="datetime.datetime"),
```

Comment:
`NameExpr` should be replaced with `RefExpr`. `NameExpr` will only handle the following code:

```python
from datetime import datetime

datetime.utcnow()  # Replace `utcnow()` with ...
```

`NameExpr` will not emit an error for the following code though:

```python
import datetime

# no error emitted
datetime.datetime.utcnow()
```

The reason being is that `datetime.datetime` is a member expression (ie, has a dot), whereas the `datetime` in the first example is a single identifier. Using a `RefExpr` covers both cases.

I will make sure to update the docs to explain when and where to use `NameExpr` vs `RefExpr`.

Reply:
Sure, good catch! 👍